numpy/_core/src/umath/stringdtype_ufuncs.cpp

```diff
@@ -1579,42 +1481,28 @@ center_ljust_rjust_resolve_descriptors(
 {
     PyArray_StringDTypeObject *input_descr = (PyArray_StringDTypeObject *)given_descrs[0];
     PyArray_StringDTypeObject *fill_descr = (PyArray_StringDTypeObject *)given_descrs[2];
+    PyArray_StringDTypeObject *common_descr = common_instance(input_descr, fill_descr);
 
-    int eq_res = _eq_comparison(input_descr->coerce, fill_descr->coerce,
-                                input_descr->na_object, fill_descr->na_object);
-
-    if (eq_res < 0) {
-        return (NPY_CASTING)-1;
-    }
-
-    if (eq_res != 1) {
-        PyErr_SetString(PyExc_TypeError,
-                        "Can only do text justification operations with equal"
-                        "StringDType instances.");
+    if (common_descr == NULL) {
         return (NPY_CASTING)-1;
     }
 
-    Py_INCREF(given_descrs[0]);
-    loop_descrs[0] = given_descrs[0];
-    Py_INCREF(given_descrs[1]);
-    loop_descrs[1] = given_descrs[1];
-    Py_INCREF(given_descrs[2]);
-    loop_descrs[2] = given_descrs[2];
+    loop_descrs[0] = NPy_NewDescrRef(given_descrs[0]);
+    loop_descrs[1] = NPy_NewDescrRef(given_descrs[1]);
+    loop_descrs[2] = NPy_NewDescrRef(given_descrs[2]);
 
     PyArray_Descr *out_descr = NULL;
 
     if (given_descrs[3] == NULL) {
         out_descr = (PyArray_Descr *)new_stringdtype_instance(
```

Comment:
As above, why not use `common_instance` directly?

Reply:
That's a good point but it doesn't work because we need to use `common_instance` for the error checking even in the other branch of the enclosing if statement here. Instead I refactored the compatibility checking into its own function and used that, avoiding creating new objects or calling `common_instance`. Thank you for pointing this out, I used `common_instance` without thinking about how it creates a `PyObject` unnecessarily.